docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+In this method, a Plugin designer should perform any cleanup for their plugin.
+        However, if the plugin is an input-type tool (it has no incoming connections),
+        processing (record generation) should occur here.
+
+        Note: A tool with an optional input anchor and no incoming connections should
+        also write any records to output anchors here.
+        """
+        try:
+            if self.MODE == 'DATA':
+                self.setup_data()
+            elif self.MODE == 'TRAIN':
+                fn_args = self.get_model_args()
+                # TODO: Update this to its own function.
+                # NOTE: Need to remove test data pushes
+                df = pd.DataFrame({"Results": [0.0, 0.0]})
+                batch_to_send = pa.RecordBatch.from_pandas(df=df, preserve_index=False)
+                self.provider.write_to_anchor("Output", batch_to_send)
+                q = Queue()
+                p = Process(target=self.create_and_save_model, args=(q, *fn_args))
+                p.start()
+                history = q.get()
+                p.join()
+                history_dict = history.history
+                conf = self.provider.full_config
+                self.info("Setting new history")
+
+                # Pot, meet kettle. Need another method or opportunity for Additional Execercise section?
+                ui_history = {}
+
+                ui_history["trainingLoss"] = history_dict["loss"]
+                ui_history["trainingBinaryAccuracy"] = history_dict["binary_accuracy"]
+                ui_history["validationLoss"] = history_dict["val_loss"]
+                ui_history["validationBinaryAccuracy"] = history_dict["val_binary_accuracy"]
+                conf['Configuration']["modelEvaluation"]['history'] = ui_history
+                self.provider.save_full_config(conf)
+            elif self.MODE == 'EVAL':
+                model = tf.keras.saving.load_model(self.provider.tool_config["modelConfig"]["modelName"], custom_objects={'custom_standardization': custom_standardization})
+            elif self.MODE == 'PREVIEW':
+                self.send_preview_data()
+```
+
+
+The majority of the above speaks for itself at this point in the guide.
+Our most notable (conceptual) addition is likely the `MultiProcessing` section:  
+
+```python
+q = Queue()
+p = Process(target=self.create_and_save_model, args=(q, *fn_args))
+p.start()
+history = q.get()
+p.join()
+```
+
+Note that this is "just enough" to prevent blocking. An incredible amount of resources are available online and on Python's official documentation. For our purposes, know we're using the `Process(...)` to wrap up our expensive compute task - allowing us to avoid blocking the Python server IO.
+
+Now the backend can generate, train, and deploy/export a new model! All that's left now is our final step, allowing the User to use the "production" model to predict data in a workflow!
```

Comment:
TW review:

Note that this is just enough to prevent blocking. An incredible amount of resources are available online and on Python's official documentation. For these purposes, know we use the `Process(...)` to wrap up the expensive compute task—this allows us to avoid blocking the Python server IO.

Now the back end can generate, train, and deploy a new model! All that's left now is the final step, allowing the user to use the "production" model to predict data in a workflow!